File: clients/java/src/main/java/com/hook0/client/QueryParameter.java
package com.hook0.client;

/**
 * One name and value of a query string, before either is escaped.
 *
 * <p>A list of these rather than a map: the document declares parameters a request may carry more than once, and a map
 * would keep whichever one was written last.
 *
 * @param name what the parameter travels under
 * @param value what it carries
 */
public record QueryParameter(String name, String value) {}

File: clients/kotlin/src/main/kotlin/com/hook0/kotlin/QueryParameter.kt
package com.hook0.kotlin

/**
 * One name and value of a query string, before either of them is escaped.
 *
 * A list of these rather than a map: the API declares parameters a request may carry more than
 * once, and a map would keep the last one and drop the rest without saying so.
 *
 * @property name what the parameter is called on the wire
 * @property value what it carries
 */
data class QueryParameter(val name: String, val value: String)
